File: src/my_cpp_pkg/src/smartphone.cpp
#include "rclcpp/rclcpp.hpp"

// ros2 interface show example_interfaces/msg/String
// string.data <- name of the variable in string class. 
#include "example_interfaces/msg/string.hpp"

using namespace std::placeholders;

#define MSG_BUF_SIZE 10
#define TOPIC_NAME "robot_news"

class SmartphoneNode : public rclcpp::Node
{
    public:
        SmartphoneNode() : Node( "smartphone" )
        {
            this->subscriber_  = this->create_subscription<example_interfaces::msg::String>( TOPIC_NAME,
                                MSG_BUF_SIZE,
                                std::bind( &SmartphoneNode::callbackRobotNews, this,
                                _1 ) );  // If using namescape std::placeholders.
                                // std::placeholders::_1 ) );
            RCLCPP_INFO( this->get_logger(), "Smartphone has been started." );
        };

    private:
    rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber_;

        void callbackRobotNews( const example_interfaces::msg::String::SharedPtr msg )
        {
            RCLCPP_INFO( this->get_logger(), "%s", msg->data.c_str() );
        };
};

int main( int argc, char **argv )
{
    rclcpp::init( argc, argv );
    auto node = std::make_shared<SmartphoneNode>();
    rclcpp::spin( node );
    rclcpp::shutdown();
    return 0;
};

File: src/my_py_pkg/my_py_pkg/smartphone.py
#!/usr/bin/env python3
#

import rclpy
from rclpy.node import Node

# ros2 interface show example_interfaces/msg/String
# string.data <- name of the variable in string class. 
from example_interfaces.msg import String

MSG_BUF_SIZE = 10
TOPIC_NAME = "robot_news"

class SmartphoneNode( Node ):
    def __init__( self ):
        super().__init__( "smartphone" )
        self.subscriber_ = self.create_subscription( String, TOPIC_NAME,
                                                     self.callback_robot_news,
                                                     MSG_BUF_SIZE )
        self.get_logger().info( "Smartphone has been started." )

    def callback_robot_news( self, msg: String ):
        self.get_logger().info( msg.data )

def main( args = None ):
    rclpy.init( args = args )
    node = SmartphoneNode()
    rclpy.spin( node )
    rclpy.shutdown()


if __name__ == "__main__":
    main()
